include/pybind11/pytypes.h

```diff
@@ -100,9 +105,11 @@ class handle : public detail::object_api<handle> {
 class object : public handle {
 public:
     object() = default;
+    // Derived classes may override this constructor for Python type conversions
+    object(handle h, bool borrowed) : handle(h) { if (borrowed) inc_ref(); }
+    // This one must be inherited as is -- it's always just a pure pointer assignment
+    object(handle h, bool borrowed, detail::noconvert) : object(h, borrowed) { }
```

Comment:
I don't understand why this special ``detail::noconvert`` constructor is needed. Isn't having a ``true/false`` argument enough to uniquely specify what is meant? (i.e. just borrow/steal a reference, but don't run any conversions).

Or is there some version of this constructor, which additional does something to the object? In that case, it may be worth rethinking that.